Allow `Rvalue::Repeat` to return true in `rvalue_creates_operand` too

scottmcm opened the pull request:
The conversation in https://github.com/rust-lang/rust/pull/143502#discussion_r2189410911 made be realize how easy this is to handle, since the only possibilty is ZSTs -- everything else ends up with the destination being `LocalKind::Memory` and thus doesn't call `codegen_rvalue_operand` at all.

This gets us perilously close to a world where `rvalue_creates_operand` only ever returns true.  (See rust-lang/rust#143860 for more.)

scottmcm committed:
Allow `Rvalue::Repeat` to return true in `rvalue_creates_operand` too

The conversation in 143502 made be realize how easy this is to handle, since the only possibilty is ZSTs -- everything else ends up with the destination being `LocalKind::Memory` and thus doesn't call `codegen_rvalue_operand` at all.

This gets us perilously close to a world where `rvalue_creates_operand` only ever returns true.  I'll try out such a world next :)

matthiaskrgr commented:
@bors r-  
https://github.com/rust-lang/rust/pull/144138#issuecomment-3090135328

scottmcm committed:
Split repeat-operand codegen test

Looks like the output it's looking for can be in different orders, so separate tests will fix that.
